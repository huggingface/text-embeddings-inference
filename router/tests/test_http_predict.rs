mod common;

use crate::common::{start_server, Score};
use anyhow::Result;
use insta::internals::YamlMatcher;
use serde::{Deserialize, Serialize};
use serde_json::json;
use text_embeddings_backend::DType;

#[derive(Serialize, Deserialize, Debug, PartialEq)]
pub struct SnapshotPrediction {
    score: Score,
    label: String,
}

#[tokio::test]
#[cfg(feature = "http")]
async fn test_predict() -> Result<()> {
    start_server(
        "SamLowe/roberta-base-go_emotions".to_string(),
        None,
        DType::Float32,
    )
    .await?;

    let request = json!({
        "inputs": "test"
    });

    let client = reqwest::Client::new();
    let res = client
        .post("http://0.0.0.0:8090/predict")
        .json(&request)
        .send()
        .await?;

    let predictions_single = res.json::<Vec<SnapshotPrediction>>().await?;
    let matcher = YamlMatcher::<Vec<SnapshotPrediction>>::new();
    insta::assert_yaml_snapshot!("predictions_single", predictions_single, &matcher);

    let request = json!({
        "inputs": vec![
            vec!["test"],
            vec!["test"],
            vec!["test"],
            vec!["test"],
            vec!["test"],
        ],
    });

    let client = reqwest::Client::new();
    let res = client
        .post("http://0.0.0.0:8090/predict")
        .json(&request)
        .send()
        .await?;

    let predictions_batch = res.json::<Vec<Vec<SnapshotPrediction>>>().await?;
    let matcher = YamlMatcher::<Vec<Vec<SnapshotPrediction>>>::new();
    insta::assert_yaml_snapshot!("predictions_batch", predictions_batch, &matcher);

    for predictions in &predictions_batch {
        assert_eq!(predictions, &predictions_single);
    }

    Ok(())
}
