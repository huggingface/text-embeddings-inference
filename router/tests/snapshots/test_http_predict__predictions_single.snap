---
source: router/tests/test_http_predict.rs
expression: predictions_single
---
- score: 0.9972132
  label: neutral
- score: 0.0007965705
  label: approval
- score: 0.00029795355
  label: annoyance
- score: 0.00020120452
  label: admiration
- score: 0.00019649488
  label: realization
- score: 0.00013194351
  label: excitement
- score: 0.000102370424
  label: disappointment
- score: 0.000100379
  label: disgust
- score: 0.000099765304
  label: anger
- score: 0.000085338215
  label: disapproval
- score: 0.00007990062
  label: joy
- score: 0.00007926568
  label: curiosity
- score: 0.00007919242
  label: amusement
- score: 0.000066899396
  label: confusion
- score: 0.0000655515
  label: optimism
- score: 0.000063790605
  label: love
- score: 0.00006372412
  label: sadness
- score: 0.00004477469
  label: fear
- score: 0.000041168856
  label: desire
- score: 0.000040866962
  label: surprise
- score: 0.000033375803
  label: caring
- score: 0.000027021126
  label: gratitude
- score: 0.000025252299
  label: embarrassment
- score: 0.000016195578
  label: pride
- score: 0.000013492137
  label: relief
- score: 0.000013144917
  label: grief
- score: 0.000011318838
  label: nervousness
- score: 0.0000098800765
  label: remorse
